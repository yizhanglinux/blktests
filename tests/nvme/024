#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test NVMe smart-log command on NVMeOF with a file-backed ns.

. tests/nvme/rc

DESCRIPTION="test NVMe smart-log command on NVMeOF file-backed ns"
QUICK=1

requires() {
	_nvme_requires
	_have_loop
	_require_nvme_trtype_is_fabrics
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local ns

	_nvmet_target_setup --blkdev file

	_nvme_connect_subsys

	ns=$(_find_nvme_ns ${def_subsys_uuid})

	if ! nvme smart-log "/dev/${ns}" >> "$FULL" 2>&1; then
		echo "ERROR: smart-log file-ns failed"
	fi

	_nvme_disconnect_subsys >> "$FULL" 2>&1

	_nvmet_target_cleanup

	echo "Test complete"
}
