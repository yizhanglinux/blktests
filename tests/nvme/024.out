Running nvme/024
Test complete
